syntax = "proto3";

service GrpcChat {
  rpc Register(AuthRequest) returns (AuthResponse) {}
  rpc Auth(AuthRequest) returns (AuthResponse) {}

  rpc PostMessage(PostMessageRequest) returns (PostMessageResponse) {}
  rpc GetStreamMessage(GetStreamMessageRequest)
      returns (stream GetStreamMessageResponse) {}
}

message AuthRequest {
  string username = 1;
  string password = 2;
}

message AuthResponse { string token = 1; }

message PostMessageRequest {
  string token = 1;
  string text = 2;
}

message PostMessageResponse { bool ok = 1; }

message GetStreamMessageRequest { string token = 1; }

message GetStreamMessageResponse {
  string user = 1;
  string text = 2;
}